src/main/scala/zio/keeper/ProtocolDef.scala

```diff
@@ -0,0 +1,70 @@
+package zio.keeper
+
+import zio.{ Chunk, IO, ZIO }
+import zio.keeper.Error.ProtocolDef.{
+  CannotFindSerializerForMessage,
+  CannotFindSerializerForMessageId,
+  SerializationError
+}
+import zio.keeper.ProtocolDef.MessageDef
+
+import scala.reflect.ClassTag
+
+trait ProtocolDef[A] {
+  def protocolName: String // it would be cool to have byte limit for name
+
+  def version: ProtocolDef.Version
+
+  def schema: Schema[A]
+
+  protected def message[B <: A](
+    to: B => IO[SerializationError, Chunk[Byte]],
```

Comment:
Is it possible to "promote" `to` and `from` to trait-level methods and remove notion of `Schema`?

Reply:
because we need to identify message type somehow it would be harder to put this in protocol.